tests/components/myuplink/test_config_flow.py

```diff
@@ -81,3 +69,98 @@ async def test_full_flow(
 
     assert len(hass.config_entries.async_entries(DOMAIN)) == 1
     assert len(mock_setup.mock_calls) == 1
+
+
+async def test_flow_reauth(
+    hass: HomeAssistant,
+    hass_client_no_auth: ClientSessionGenerator,
+    aioclient_mock: AiohttpClientMocker,
+    current_request_with_host: None,
+    setup_credentials: None,
+    mock_config_entry: MockConfigEntry,
+    expires_at: float,
+) -> None:
+    """Test reauth step."""
+
+    NEW_SCOPE = "WRITESYSTEM READSYSTEM offline_access"
+    OLD_SCOPE = "READSYSTEM offline_access"
+    OLD_SCOPE_TOKEN = {
+        "auth_implementation": DOMAIN,
+        "token": {
+            "access_token": "Fake_token",
+            "scope": OLD_SCOPE,
+            "expires_in": 86399,
+            "refresh_token": "3012bc9f-7a65-4240-b817-9154ffdcc30f",
+            "token_type": "Bearer",
+            "expires_at": expires_at,
+        },
+    }
+    assert mock_config_entry.data["token"]["scope"] == NEW_SCOPE
+    assert hass.config_entries.async_update_entry(
+        mock_config_entry, data=OLD_SCOPE_TOKEN
+    )
+    assert mock_config_entry.data["token"]["scope"] == OLD_SCOPE
+
+    mock_config_entry.add_to_hass(hass)
```

Comment:
Adding the config entry to hass should be done before updating the config entry. Add it already in the fixture for the mock config entry